backend/fms_core/tests/test_resources.py

```diff
@@ -595,3 +599,57 @@ def test_invalid_container_rename(self):
                 self._test_invalid_rename_template(rf, err)
 
             transaction.savepoint_rollback(s)
+
+    def create_projects(self):
+        self.project1, _ = Project.objects.get_or_create(name="TestProject1", status="Ongoing")
+        self.project2, _ = Project.objects.get_or_create(name="TestProject2", status="Ongoing")
+
+    def link_projects_to_samples(self):
+        self.load_samples()
+        self.create_projects()
+
+        with reversion.create_revision(), open(PROJECT_LINK_SAMPLES_CSV) as sf:
+            s = Dataset().load(sf.read())
+            self.pr.import_data(s, raise_errors=True)
+
+            reversion.set_comment("Linked projects with samples")
+
+    def test_link_projects_to_samples_import(self):
+        self.link_projects_to_samples()
+        sample1 = Sample.objects.get(name="sample1", container__barcode="tube001")
+        sample2 = Sample.objects.get(name="sample2", container__barcode="tube002")
+
+        # Test basic import success
+        self.assertEqual(len(SampleByProject.objects.all()), 2)
+
+        # Test parent record auto-generation
+        self.assertTrue(SampleByProject.objects.filter(sample=sample1, project=self.project1).exists())
+        self.assertTrue(SampleByProject.objects.filter(sample=sample2, project=self.project2).exists())
+
+    def unlink_projects_to_samples(self):
```

Comment:
Tests methods should start with "test_" .  The real test seem to have been commented out below ... Any reason ?

Reply:
Not sure that this method 'unlink_projects_to_samples' is ever called (it is in your commented block).

If it is a test case by itself, indeed it should start with "test_"
If it is a helper method (such as in load_samples, load_containers, etc) it should indeed not start with "test_"

So from what I understand from your code, you just forgot to uncomment your test_unlink_proejcts_to_sample function :)